src/twisted/names/client.py

```diff
@@ -439,6 +451,43 @@ def _cbLookupZone(self, result, connector):
 
 
 
+class IPv6Resolver(Resolver):
+    """
+    Selects only IPv6 resolv.conf entries.
+    """
+    def maybeAddNameserver(self, nameserver, servers):
+        resolver = (nativeString(nameserver), dns.PORT)
+        if b':' in nameserver:
```

Comment:
[`isIPAddress`](https://twistedmatrix.com/documents/16.6.0/api/twisted.internet.abstract.html#isIPAddress) and [`isIPv6Address`](https://twistedmatrix.com/documents/16.6.0/api/twisted.internet.abstract.html#isIPv6Address) might be helpful here.